### ISSUE TITLE
album exclude does not delete html file

### ISSUE THREAD

**Issue by oupala** [thumbsup/thumbsup#152]
## Bug report

```
Thumbsup version: 2.8.1
Node version: v10.15.1
NPM version: 6.4.1
Operating system: debian stretch
```

About the bug...

- generate a gallery
- then exclude a folder in the configuration, and enable the cleanup flag
- generate the gallery again: the html file of the gallery is not cleaned-up

**Comment by rprieto** [comment]
Good catch thanks! The `--clean` option only applies to generated photos and videos for now. I like the idea of cleaning up HTML files too.

One challenge is that:
- thumbsup "owns" the `./media` folder. We can delete anything that doesn't look relevant 
- thumbsup doesn't own the root folder, because it could contain other files such as a website index, a 404 page, images from a custom theme, etc

Do you have suggestions on how clean old files in a safe way?

**Comment by oupala** [comment]
In my opinion, a static generator should own all the directories where it writes files.

It should hence own the root folder either. Is it possible to change this ownership?

**Comment by rprieto** [comment]
I didn't mean "own" in the sense of Unix permissions. I meant that not *all* files in the output folder are generated by thumbsup. For example:
- I know people using thumbsup inside a site that already has other pages (hence the `--index` option to specify a different name for the root HTML page)
- I myself have extra HTML pages like a login page, 403, 404, ...

I'm keen to hear suggestions that can ensure we never cleanup any "user" files.

**Comment by oupala** [comment]
I can't see any simple way for thumbsup to manage a directory content that is also modified by something else (*something else* can be a human).

I see two solutions:

- add a flag to tell if the root folder is *owned* by thumbsup, if no then html files are never deleted, if yes then html files can be securely deleted if the album does not exist any more
- the second solution is to use a prefix for html files, the prefix can then be used by thumbsup to determine which files are under its responsibility

Does anyone has another solution?

**Comment by rprieto** [comment]
Similar to the prefix, it makes me think that I always use the `--albums-output-folder` option to keep the gallery folder tidy. One option is to always clean the `media` folder and the `albums` folder if one is specified.

**Comment by oupala** [comment]
Another option is to offer a new option to specifiy a *whitelist* parameter that can be a list of files, of a regex, or a list of files and regex so that one can specify files manually managed **not to delete**. All other files are managed by thumbsup and can be freely deleted.

The last option I see is to offer a source directory for *static files*. You can here place you login page, your error pages, and whatever html file you want, these files are automatically copied by thumbsup on build time. This way, is becomes easy for thumbsup to determine what files can be deleted from the destination directory.

**Comment by oupala** [comment]
It appears that old directories in the `media` directories, which were containing photos, are not deleted either.

**Comment by oupala** [comment]
And old zip are not deleted either.

When the source directory evolves with modification and delete, the destination directory is in the end really bigger that it should be.
